### PR TITLE
openjdk@8: upstream Linux fixes

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
```
 /usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/5/../../..lib/CoreLibraries.gmk:109: recipe for target '/tmp/openjdkA8-20210710-8467-8dlxp6/jdk8u282-ga/build/linux-x86_64-normal-server-release/jdk/lib/amd64/libverify.so' failed
/x86_64-linux-gnu/crt1.o: relocation R_X86_64_32S against `__libc_csu_fini' can not be used when making a shared object; recompile with -fPIC
/usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/crt1.o: error adding symbols: Bad value
collect2: error: ld returned 1 exit status
make[2]: *** [/tmp/openjdkA8-20210710-8467-8dlxp6/jdk8
```

**Comment by danielnachun** [comment]
This is a pretty arcane error.  I'm going to test locally in Docker and see if I can reproduce this.

**Comment by danielnachun** [comment]
I figured out why it's happening though I don't have a solution yet.  We were bypassing the compiler shims on Linux to get this to build and if I change
```
 args += %W[CC=#{ENV.cc}
                  CXX=#{ENV.cxx}
```
to 
```
 args += %W[CC=/usr/bin/cc
                  CXX=/usr/bin/c++
```
as it was in linuxbrew-core, it works fine.

Here are the gist-logs for a local build I did without the shim bypass that failed: https://gist.github.com/6ef11b3e19d088e3b7f3270c43ce5d78.

File: Formula/openjdk@8.rb [hunk 1]
```diff
@@ -92,11 +134,20 @@ def install
...
-    bin.install_symlink Dir["#{libexec}/openjdk.jdk/Contents/Home/bin/*"]
-    include.install_symlink Dir["#{libexec}/openjdk.jdk/Contents/Home/include/*.h"]
-    include.install_symlink Dir["#{libexec}/openjdk.jdk/Contents/Home/include/darwin/*.h"]
+    on_macos do
+      jdk = Dir["build/*/images/j2sdk-bundle/*"].first
+      libexec.install jdk => "openjdk.jdk"
+      bin.install_symlink Dir["#{libexec}/openjdk.jdk/Contents/Home/bin/*"]
+      include.install_symlink Dir["#{libexec}/openjdk.jdk/Contents/Home/include/*.h"]
+      include.install_symlink Dir["#{libexec}/openjdk.jdk/Contents/Home/include/darwin/*.h"]
+    end
```
Comment (carlocab) [inline Formula/openjdk@8.rb:143]:
```suggestion
    on_macos do
      jdk = Dir["build/*/images/j2sdk-bundle/*"].first
      libexec.install jdk => "openjdk.jdk"
      bin.install_symlink Dir[libexec/"openjdk.jdk/Contents/Home/bin/*"]
      include.install_symlink Dir[libexec/"openjdk.jdk/Contents/Home/include/*.h"]
      include.install_symlink Dir[libexec/"openjdk.jdk/Contents/Home/include/darwin/*.h"]
      man1.install_symlink Dir[libexec/"openjdk.jdk/Contents/Home/man/man1/*"]
    end
```

File: Formula/openjdk@8.rb [hunk 2]
```diff
@@ -92,11 +134,20 @@ def install
...
+
+    on_linux do
+      libexec.install Dir["build/*/images/j2sdk-image/*"]
+      bin.install_symlink Dir["#{libexec}/bin/*"]
+      include.install_symlink Dir["#{libexec}/include/*.h"]
+      include.install_symlink Dir["#{libexec}/include/linux/*.h"]
```
Comment (carlocab) [inline Formula/openjdk@8.rb:149]:
```suggestion
      libexec.install Dir["build/*/images/j2sdk-image/*"]
      bin.install_symlink Dir[libexec/"bin/*"]
      include.install_symlink Dir[libexec/"include/*.h"]
      include.install_symlink Dir[libexec/"include/linux/*.h"]
      man1.install_symlink Dir[libexec/"man/man1/*"]
```
Let's add the changes from the other PRs here too.

**Comment by jonchang** [comment]
Pushed a fix that works for me locally on my Linux box, so let's hope this works.

**Comment by jonchang** [comment]
:tada:

**Comment by danielnachun** [comment]
Wow, that was not obvious at all!  Thank you so much for figuring this out @jonchang.

**Comment by jonchang** [comment]
Linux linker errors are always a bundle of fun. Note that there are a few patches in openjdk@8 HEAD that will mean that we will need to look at this again when it updates, by removing a few of our inreplaces here.

**Pull request merged**